runtime-modules/content-directory/src/lib.rs

```diff
@@ -0,0 +1,1645 @@
+        ) -> dispatch::Result {
+            let account_id = ensure_signed(origin)?;
+
+            Self::ensure_known_entity_id(entity_id)?;
+
+            let (entity, access_level) = Self::get_entity_and_access_level(account_id, entity_id, actor)?;
+
+
+            let current_property_value_vec =
+            Self::get_property_value_vec(&entity, in_class_schema_property_id)?;
+
+            Self::ensure_class_property_type_unlocked_for(
+                entity.class_id,
+                in_class_schema_property_id,
+                access_level,
+            )?;
+
+            let entities_rc_to_decrement = current_property_value_vec
+                .get_vec_value()
+                .get_involved_entities();
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Clear property value vector:
+            <EntityById<T>>::mutate(entity_id, |entity| {
+                if let Some(PropertyValue::Vector(current_property_value_vec)) =
+                    entity.values.get_mut(&in_class_schema_property_id)
+                {
+                    current_property_value_vec.vec_clear();
+                }
+                if let Some(entities_rc_to_decrement) = entities_rc_to_decrement {
+                    Self::decrement_entities_rc(&entities_rc_to_decrement);
+                }
+            });
+
+            Ok(())
+        }
+
+        pub fn remove_at_entity_property_vector(
+            origin,
+            actor: Actor<T>,
+            entity_id: T::EntityId,
+            in_class_schema_property_id: PropertyId,
+            index_in_property_vec: VecMaxLength,
+            nonce: T::Nonce
+        ) -> dispatch::Result {
+            let account_id = ensure_signed(origin)?;
+
+            Self::ensure_known_entity_id(entity_id)?;
+
+            let (entity, access_level) = Self::get_entity_and_access_level(account_id, entity_id, actor)?;
+
+            let current_property_value_vec =
+            Self::get_property_value_vec(&entity, in_class_schema_property_id)?;
+
+            Self::ensure_class_property_type_unlocked_for(
+                entity.class_id,
+                in_class_schema_property_id,
+                access_level,
+            )?;
+
+            // Ensure property value vector nonces equality to avoid possible data races,
+            // when performing vector specific operations
+            current_property_value_vec.ensure_nonce_equality(nonce)?;
+            current_property_value_vec
+                .ensure_index_in_property_vector_is_valid(index_in_property_vec)?;
+            let involved_entity_id = current_property_value_vec
+                .get_vec_value()
+                .get_involved_entities()
+                .map(|involved_entities| involved_entities[index_in_property_vec as usize]);
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Remove property value vector
+            <EntityById<T>>::mutate(entity_id, |entity| {
```

Comment:
Is nonce being updated?

Reply:
Yes, nonce have been updated

Reply:
You can find it incapsulated inside vector specific `PropertyValue` operation, like `insert_at`, `remove_at`